### PR TITLE
FEATURE: Adding re-usable crank_checks function

### PR DESCRIPTION
**Pull request by coachchucksol**
The new `crank_checks` function does all of the checks needed across each crank. Now with the new addition of looking for mismatched indices.

### CODE REVIEW

File: programs/steward/src/instructions/idle.rs [hunk 1]
```diff
@@ -14,6 +16,10 @@ pub struct Idle<'info> {
         bump
     )]
     pub state_account: AccountLoader<'info, StewardStateAccount>,
+
+    /// CHECK: Account owner checked, account type checked in get_validator_stake_info_at_index
```
Comment (ebatsell) [inline programs/steward/src/instructions/idle.rs:20]: nit: account type checked in `get_validator_list_length`, and owner is not checked but address is verified by being set in config

File: programs/steward/src/utils.rs [hunk 2]
```diff
@@ -8,7 +8,47 @@ use spl_stake_pool::{
     state::{StakeStatus, ValidatorListHeader, ValidatorStakeInfo},
 };
 
-use crate::{errors::StewardError, Config, Delegation};
+use crate::{errors::StewardError, Config, Delegation, StewardStateAccount, StewardStateEnum};
+
+pub fn crank_check(
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply programs/steward/src/utils.rs:13]: Done

File: programs/steward/src/state/steward_state.rs [hunk 3]
```diff
@@ -249,6 +249,7 @@ pub const POST_LOOP_IDLE: u32 = 1 << 6;
 /// once for any validators that still need to be removed
 /// when there are no validators to remove from the pool, the operation continues
 /// and this condition is not checked again
+/// DEPRECATED: This flag is no longer used
 pub const CHECKED_VALIDATORS_REMOVED_FROM_LIST: u32 = 1 << 16;
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply programs/steward/src/state/steward_state.rs:253]: Done

File: programs/steward/src/instructions/idle.rs (hunk 1 above)
Comment (coachchucksol) [reply programs/steward/src/instructions/idle.rs]: Done

**Pull request merged**